python_ta/contracts/__init__.py

```diff
@@ -184,6 +184,11 @@ def _check_function_contracts(wrapped, instance, args, kwargs):
         if param in annotations:
             try:
                 _debug(f"Checking type of parameter {param} in call to {wrapped.__qualname__}")
+                if type(arg) is int and annotations[param] is float:
```

Comment:
It's not great to duplicate this code in two different places. Instead, create a new helper function that wraps both the check you added and the call to `check_type` itself. The new function should take the same arguments as `check_type` itself.

Reply:
I've now updated the helper function to raise a TypeError.